build-support/docker/centos6/Dockerfile

```diff
@@ -4,18 +4,38 @@
 # Use centos6 for compatibility with glibc 2.12.
 FROM centos:6
 
-# Install python 2.7.13, (more) modern gcc, and a JDK.
+# Install a (more) modern gcc, a JDK, and dependencies for installing Python through Pyenv.
+# Also install sqlite-devel because python 3 errors on startup if not -- see
+# https://stackoverflow.com/a/39907500/2518889.
 RUN yum -y update
+# TODO: figure out why this needs to be installed first for /usr/bin/scl to work!
 RUN yum install -y centos-release-scl
 RUN yum install -y \
-  devtoolset-7-gcc \
-  devtoolset-7-gcc-c++ \
-  git \
-  java-1.8.0-openjdk-devel \
-  python27 \
-  libffi \
-  libffi-devel \
-  openssl-devel
+        bzip2-devel \
+        devtoolset-7-gcc{,-c++} \
+        git \
+        java-1.8.0-openjdk-devel \
+        libffi-devel \
+        openssl-devel \
+        readline-devel \
+        sqlite-devel \
+        zlib-devel
 
-# By default, execute in an environment with python27 enabled.
-ENTRYPOINT ["/usr/bin/scl", "enable", "python27", "devtoolset-7",  "--"]
+ARG PYTHON_27_VERSION=2.7.13
+ARG PYTHON_36_VERSION=3.6.8
+# NB: PYENV_ROOT must be set for `pyenv install` to be available. This failure mode is not mentioned
+# in https://github.com/pyenv/pyenv#basic-github-checkout.
+ENV PYENV_ROOT "${PYENV_ROOT:-/pyenv-docker-build}"
```

Comment:
Because this is the base image, I don't think there is any circumstance where `PYENV_ROOT` will be predefined. If this is the case, I'd say simpler is better.

Reply:
Was going to fix that -- thanks!